cpp/src/arrow/engine/substrait/options.h

```diff
@@ -67,6 +69,16 @@ using NamedTableProvider =
     std::function<Result<compute::Declaration>(const std::vector<std::string>&)>;
 static NamedTableProvider kDefaultNamedTableProvider;
 
+using NamedTapKindMapper = std::function<Result<compute::Declaration>(
+    const std::string&, std::vector<compute::Declaration::Input>,
+    std::shared_ptr<compute::ExecNodeOptions>)>;
+static NamedTapKindMapper kDefaultNamedTapKindMapper =
+    [](const std::string& kind, std::vector<compute::Declaration::Input> inputs,
+       std::shared_ptr<compute::ExecNodeOptions> options)
+    -> Result<compute::Declaration> {
+  return compute::Declaration(kind, inputs, options);
```

Comment:
If we remove `NamedTapNodeOptions` like some of my other comments suggest then I suppose there is no more meaningful default mapper.  That being said, I don't know of any use case where this default would be correct.  Perhaps we just return an error here like we do with named tables?

Reply:
Done.